### ISSUE TITLE
heap command can't run on Ubuntu18.04 which is a docker container

### ISSUE THREAD

**Issue by JD1412** [pwndbg/pwndbg#1327]
# heap command can't run on Ubuntu18.04 which is a docker container

## My Environment

I install the pwndbg in a docker container,the Container is based on Ubuntu18.04 system

## My Problem

when I want to debug an ELF file by the heap command,for example heap,bins,Unsortedbin,vis_heap_chunks
the pwndbg give me tips like this

<img width="540" alt="image" src="https://user-images.githubusercontent.com/30342224/197388693-ac02c2a5-8217-4d2e-ad19-59fe61b79aed.png">

so I run the command 

<img width="807" alt="image" src="https://user-images.githubusercontent.com/30342224/197388727-cede1876-37a5-4fb7-9f18-9c8a7402a809.png">

then I run the heap command again 

<img width="881" alt="截屏2022-10-23 19 11 05" src="https://user-images.githubusercontent.com/30342224/197400612-de58b853-03c9-417c-966c-9f22aa25ad16.png">

I have no idea to solve the problem

**Comment by lebr0nli** [comment]
Hi, thanks for reporting the bug!
I guess the hint about the heap heuristic probably misled you.
Could you re-run the `heap` command instead of `set resolve-heap-via-heuristic on` after you `set resolve-heap-via-heuristic on` once, and test if that work for you?
Could you also `set exception-verbose on` and show the full stacktrace when you use the `heap` command?

Thanks!

**Comment by JD1412** [comment]
Oh my god,I found an error in my question,I paste the same picture in different step. I have corrected that mistake.
I try it again.
<img width="890" alt="截屏2022-10-23 23 24 12" src="https://user-images.githubusercontent.com/30342224/197400763-bd1e7e24-1416-40fe-9e53-8bbfac348554.png">

when I run the command `set exception-verbose on`,I got this screenshot
<img width="899" alt="截屏2022-10-23 23 27 39" src="https://user-images.githubusercontent.com/30342224/197400953-a9810a89-76ff-4ed4-a46b-a6fce122af8e.png">

**Comment by lebr0nli** [comment]
@JD1412 , thanks!

I just reproduced this bug on Ubuntu 18.04 docker container now, the error seems occurred because the version of the GDB in the container is less than 9.x(the GDB version in my 18.04 container is 8.1.1), but `gdb.look_up_static_symbol`  required GDB 9.x. Will try to let it supports GDB 8.x without this error, thanks for pointing out this issue!

**Comment by JD1412** [comment]
I run the command `bugreport --run-browser` just now, and what you said is the same as my 'bugreport'. 
I understand the reason of the bug probably . I try to use another version of the gdb temporary. 
This has been bothering me for a long time. Thank you very much. Thanks a million.

**Comment by disconnect3d** [comment]
@JD1412 you can always compile the latest GDB yourself and just use that:

```
wget https://ftp.gnu.org/gnu/gdb/gdb-12.1.tar.xz
tar -xvf gdb-12.1.tar.xz
cd gdb-12.1
mkdir build
cd build
../configure --disable-nls --disable-werror --with-system-readline --with-python=`which python3` --with-system-gdbinit=/etc/gdb/gdbinit --enable-targets=all
make -j7
```

You may need to install additional packages for that like `texinfo` or some others. Also this will build GDB in multiarch mode so that you can debug all archs supported in GDB.

(But yeah, we should and will fix this problem)

**Comment by gsingh93** [comment]
I investigated why this wasn't caught with our testing infrastructure. It turns out the Github hosted workers actually install a newer version of GDB (10.2) from a PPA according to the logs:
```
Get:4 http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu bionic/main amd64 gdb amd64 10.2-0ubuntu1~18.04~2 [3266 kB]
```
This is annoying, because the whole point of CI is to mimic an actual production system, which they don't seem to be doing here.

I'm wondering if we should switch to running all tests inside Docker, which would properly match a user's environment. Apparently these containers should be already cached on the hosted workers: https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu1804-Readme.md#cached-docker-images

The 22.04 hosted worker contains cached Docker images for all the Ubuntu versions we care about, so maybe we should switch to that for all of our tests: https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#cached-docker-images

**Comment by JD1412** [comment]
I am honored to receive your reply, I try to install GDB 10.2 manually(I try to install GDB 12.1, but I have failed), After that I can run the command `heap`,
<img width="786" alt="image" src="https://user-images.githubusercontent.com/30342224/197577800-c9304c08-e157-451d-b08b-dc9eed4866b8.png">
Thank you very much!!!
Now my gdb with `pwndbg` can run heap command normally, I'm looking forward to use the version which is fixed that problem.
Thank you.

**Comment by disconnect3d** [comment]
Afaik this has been additionally fixed in #1333 - we now fallback to `gdb.lookup_symbol` if `gdb.lookup_static_symbol` is not available (in older GDB versions).
